New entries in UI and UX -- fixing PR (#4)
js6450/docs: content/uiux/concepts/usability-heuristics/usability-heuristics.md

---
Title: 'Usability Heuristics'
Description: 'The ten usability heuristics are a set of general principles for creating effective interaction design.'
Subjects:
  - 'Web Design'
Tags:
  - 'UI'
  - 'UX'
CatalogContent:
  - 'intro-to-ui-ux'
  - 'paths/front-end-engineer-career-path'
---

The ten **usability heuristics**, developed by [Jakob Nielsen](https://www.nngroup.com/people/jakob-nielsen/) and [Don Norman](https://jnd.org/), are a set of general principles for creating effective interaction design. They help designers assess where a design is excelling and falling short, and what trade-offs are being made in a design process. The ten usability heuristics are used in the [heuristic evaluation](https://www.codecademy.com/resources/docs/uiux/heuristic-evaluation), a relatively quick and flexible method of usability testing.

![An illustrated list of Jakob Nielsen's ten usability heuristics.](https://static-assets.codecademy.com/Courses/intro-to-ui-and-ux/key-methodologies/usability-heuristics.png)

According to the [Nielsen-Norman Group](https://www.nngroup.com/articles/ten-usability-heuristics/), the ten heuristics can be defined as follows:

> 1. **Visibility of system status**: The design should always keep users informed about what is going on, through appropriate feedback within a reasonable amount of time.

Reviewer: ```suggestion
1. **Visibility of system status**: The design should always keep users informed about what is going on through appropriate feedback within a reasonable amount of time.
```

> 2. **Match between system and the real world**: The design should speak the users' language. Use words, phrases, and concepts familiar to the user, rather than internal jargon. Follow real-world conventions, making information appear in a natural and logical order.

Reviewer: ```suggestion
2. **Match between system and the real world**: The design should speak the user's language. Uses words, phrases, and concepts familiar to the user, rather than internal jargon. Follows real-world conventions, making information appear in a natural and logical order.
```

3. **User control and freedom**: Users often perform actions by mistake. They need a clearly marked "emergency exit" to leave the unwanted action without having to go through an extended process.
4. **Consistency and standards**: Users should not have to wonder whether different words, situations, or actions mean the same thing. Follow platform and industry conventions.
5. **Error prevention**: Good error messages are important, but the best designs carefully prevent problems from occurring in the first place. Either eliminate error-prone conditions, or check for them and present users with a confirmation option before they commit to the action.
6. **Recognition rather than recall**: Minimize the user's memory load by making elements, actions, and options visible. The user should not have to remember information from one part of the interface to another. Information required to use the design (e.g. field labels or menu items) should be visible or easily retrievable when needed.

> 7. **Flexibility and efficiency of use**: Shortcuts — hidden from novice users — may speed up the interaction for the expert user such that the design can cater to both inexperienced and experienced users. Allow users to tailor frequent actions.

Reviewer: ```suggestion
7. **Flexibility and efficiency of use**: Shortcuts — hidden from novice users — may speed up the interaction for the expert user such that the design can cater to both inexperienced and experienced users. Allows users to tailor frequent actions.
```

8. **Aesthetic and minimalist design**: Interfaces should not contain information which is irrelevant or rarely needed. Every extra unit of information in an interface competes with the relevant units of information and diminishes their relative visibility.
9. **Help users recognize, diagnose, and recover from errors**: Error messages should be expressed in plain language (no error codes), precisely indicate the problem, and constructively suggest a solution.
10. **Help and documentation**: It’s best if the system doesn’t need any additional explanation. However, it may be necessary to provide documentation to help users understand how to complete their tasks.

> The [Nielsen-Norman Group's article about the ten usability heuristics](https://www.nngroup.com/articles/ten-usability-heuristics/) has more examples and tips about each heuristic. The first iteration of the ten usability heuristics was defined by Rolf Molich and Jakob Nielsen in 1990, and the heuristics themselves have remained the same since 1994, with some minor updates to the article.

Reviewer: ```suggestion
The [Nielsen-Norman Group's article about the ten usability heuristics](https://www.nngroup.com/articles/ten-usability-heuristics/) has more examples and tips about each heuristic. The first iteration of the ten usability heuristics was defined by Rolf Molich and Jakob Nielsen in 1990. The heuristics themselves have remained the same since 1994 with some minor updates to the article.
```
